monitor/monitor.go

```diff
@@ -77,14 +77,20 @@ func (m *Monitor) Run(npages int) {
 				break
 			}
 		}
-		if todo > 0 {
+		if todo > 0 && numListeners() > 0 {
```

Comment:
I would revert this change. Even if you have no listeners you must keep reading from the perf buffer, otherwise you'll start losing events and incorrectly report a high lost event count when the first reader connects.
Plus, it's always interesting to collect and report the "lost event" metric even when there are no readers.

However, you could optimize the no-listener case by not decoding the monitor events, just read and discard them.

Reply:
You want to disconnect from the perf ring buffer to stop triggering event generation in the data path. We don't want to pay the datapath cost as a penalty if no-one is listening.